### PR TITLE
docs(tuto): add github actions tutorial

### PR DESCRIPTION
**Pull request by antoinechampion**
### Your checklist for this pull request

- [X] Check that the commit messages match our requested structure.
- [X] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description
Tutorial for using Github Actions along with Scaleway Container Registry. See [related issue](https://github.com/scaleway/docs-content/issues/1227).

### CODE REVIEW

**Review by bene2k1 — APPROVED** [review]

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  - You have [created a working Dockerfile in the root of that repository](https://docs.docker.com/engine/reference/builder/) in that organization
+</Message>
+
+## Setting up your Scaleway secrets in Github
+
+In order for your Github Action to interact with your Scaleway Container Registry, it is necessary to provide the required data to your repository. The first step is to store all the necessary information in Github secrets.
+
+1. Go to the **Settings** page of your repository. In the "Secrets and variables" menu, select **Actions**.
+
+2. We will create a first secret holding your Scaleway API key.
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:38]:
```suggestion
2. Create a first secret holding your Scaleway API key.
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    Click **New repository secret**. In the "name" field, enter `SCALEWAY_API_KEY`, and paste your API secret key in the "value" field. The Access Key ID will not be used here.
+    <Message type="important">
+        Beware to not leave any leading or trailing spaces in the "value" field.
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:41]:
```suggestion
        Make sure not to leave any leading or trailing spaces in the "value" field.
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    </Message>
+
+3. We will create a second secret holding your container registry endpoint. You can find your endpoint in your Scaleway Container Registry settings:
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:44]:
```suggestion
3. Create a second secret holding your container registry endpoint. You can find your endpoint in your Scaleway Container Registry settings:
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    <Lightbox src="scaleway-container-registry-settings.webp" alt="" />
+
+    Click **New repository secret** again. In the "name" field, enter `CONTAINER_REGISTRY_ENDPOINT`, and paste your endpoint in the "value" field.
+
+## Creating your Github Action workflow
+
+We will create a GitHub Actions workflow that you can trigger, which is an automated sequence of steps to build and deploy your Docker image.
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:51]:
```suggestion
Next we create a GitHub Actions workflow that you can trigger, which is an automated sequence of steps to build and deploy your Docker image.
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  - You have [the member role in a Github organization](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization)
+  - You have [created a Github repository](https://docs.github.com/en/get-started/quickstart/create-a-repo) in that organization
+  - You have [created a working Dockerfile in the root of that repository](https://docs.docker.com/engine/reference/builder/) in that organization
+</Message>
+
+## Setting up your Scaleway secrets in Github
+
+In order for your Github Action to interact with your Scaleway Container Registry, it is necessary to provide the required data to your repository. The first step is to store all the necessary information in Github secrets.
+
+1. Go to the **Settings** page of your repository. In the "Secrets and variables" menu, select **Actions**.
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:36]:
```suggestion
1. Go to the **Settings** page of your Github repository. In the "Secrets and variables" menu, select **Actions**.
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    Click **New repository secret** again. In the "name" field, enter `CONTAINER_REGISTRY_ENDPOINT`, and paste your endpoint in the "value" field.
+
+## Creating your Github Action workflow
+
+We will create a GitHub Actions workflow that you can trigger, which is an automated sequence of steps to build and deploy your Docker image.
+
+### Creating the workflow
+
+Github workflows are described by [YAML](https://en.wikipedia.org/wiki/YAML) files located in the `.github/workflows` directory from your repository.
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:55]:
```suggestion
Github workflows are described by [YAML](https://en.wikipedia.org/wiki/YAML) files located in the `.github/workflows` directory of your repository.
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+3. `docker build` is used to build a Docker image with the code from the repository. The resulting image is tagged with the container registry endpoint and the name of the image, which is `my-docker-image` in this case.
+4. `docker push` is used to push the Docker image to the Scaleway Container Registry.
+
+<Message type="note">
+    The `on.workflow_dispatch` node in the workflow YAML tree specifies that this workflow can be run manually in Github, which is the path that we chose in this tutorial. There are [many other events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) that you can choose from to trigger your workflow, such as a push or a merge.
+</Message>
+
+### Launch the workflow
+
+The workflow is now ready to get deployed. Commit and push your `build.yml` file, then get back to your repository's home page on Github.
```
Comment (RoRoJ) [inline tutorials/use-container-registry-github-actions/index.mdx:94]:
```suggestion
The workflow is now ready to be deployed. Commit and push your `build.yml` file, then go back to your repository's home page on Github.
```

**Review by antoinechampion — COMMENTED** [review]
@RoRoJ Thanks for the review. Updated the PR with your suggested changes.

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  - You have [the member role in a Github organization](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization)
+  - You have [created a Github repository](https://docs.github.com/en/get-started/quickstart/create-a-repo) in that organization
+  - You have [created a working Dockerfile in the root of that repository](https://docs.docker.com/engine/reference/builder/) in that organization
+</Message>
+
+## Setting up your Scaleway secrets in Github
+
+In order for your Github Action to interact with your Scaleway Container Registry, it is necessary to provide the required data to your repository. The first step is to store all the necessary information in Github secrets.
+
+1. Go to the **Settings** page of your Github repository. In the "Secrets and variables" menu, select **Actions**.
```
Comment (ldecarvalho-doc) [inline tutorials/use-container-registry-github-actions/index.mdx:36]:
```suggestion
1. Go to the **Settings** page of your Github repository. In the **Secrets and variables** menu, select **Actions**.
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+## Creating your Github Action workflow
+
+Next we create a GitHub Actions workflow that you can trigger, which is an automated sequence of steps to build and deploy your Docker image.
+
+### Creating the workflow
+
+Github workflows are described by [YAML](https://en.wikipedia.org/wiki/YAML) files located in the `.github/workflows` directory of your repository.
+In your repository root, create the `.github` folder and the `workflows` sub-folder, then create a file `.github/workflows/build.yml`.
+
+```
```
Comment (ldecarvalho-doc) [inline tutorials/use-container-registry-github-actions/index.mdx:58]:
```suggestion
```yml
```

File: tutorials/use-container-registry-github-actions/index.mdx [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        username: nologin
+        password: ${{ secrets.SCALEWAY_API_KEY }}
+        registry: ${{ secrets.CONTAINER_REGISTRY_ENDPOINT }}
+    - name: Build the Docker image
+      run: docker build . -t ${{ secrets.CONTAINER_REGISTRY_ENDPOINT }}/my-docker-image
+    - name: Push the Docker Image
+      run: docker push ${{ secrets.CONTAINER_REGISTRY_ENDPOINT }}/my-docker-image
+```
+
+This workflow has one job called "build", which is set to run on the latest version of Ubuntu. The job includes four steps:
```
Comment (ldecarvalho-doc) [inline tutorials/use-container-registry-github-actions/index.mdx:81]:
```suggestion
This workflow has one job called **build**, which is set to run on the latest version of Ubuntu. The job includes four steps:
```

**Pull request merged**